app/controllers/concerns/recovery_code_concern.rb

```diff
@@ -0,0 +1,12 @@
+module RecoveryCodeConcern
+  delegate :active_profile, to: :current_user
+
+  def create_new_code
+    if active_profile.present?
+      Pii::ReEncryptor.new(current_user, user_session).perform
+      active_profile.recovery_code
```

Comment:
I don't think returning a value that matters violates SRP. If a class is meant to generate a recovery code, why can't it also return it? I think the problem here is that the `encrypt_recovery_pii` method either has too many responsibilities, or it does not fully describe what it does.

For now, I'm fine with what's in this PR, and we can refactor the profile methods later.